### ISSUE TITLE
System wide alert does not display in DSpace 8

### ISSUE THREAD

**Issue by MW3000** [DSpace/dspace-angular#3424]
## Describe the bug
System wide alert does not display in DSpace 8.x
If you turn on system wide alert and save, it might temporarily show up in your logged in session, but when you check in another browser or logout and reload the site it will not be visible for not logged in users. Sometimes it might flash up for a second and disappear again.

There also is an error message at the browser console. This is taken from https://demo.dspace.org:
```
[Error] ERROR
Error: undefined is not a valid constructor for {"type":"systemwidealert","alertId":1,"message":"Test the system wide alert feature! New try","countdownTo":null,"active":true,"_links":{"self":{"href":"https://demo.dspace.org/server/api/system/systemwidealerts/1"}}}
(anonymous function) — main.4e36a1fc2ea53912.js:1:118318
(anonymous function) — main.4e36a1fc2ea53912.js:1:1907222
(anonymous function) — main.4e36a1fc2ea53912.js:1:1903613
next — main.4e36a1fc2ea53912.js:1:1892448
(anonymous function) — main.4e36a1fc2ea53912.js:1:1907215
(anonymous function) — main.4e36a1fc2ea53912.js:1:1903613
next — main.4e36a1fc2ea53912.js:1:1892448
(anonymous function) — main.4e36a1fc2ea53912.js:1:1903613
next — main.4e36a1fc2ea53912.js:1:1892448
(anonymous function) — main.4e36a1fc2ea53912.js:1:1907215
(anonymous function) — main.4e36a1fc2ea53912.js:1:1903613
next — main.4e36a1fc2ea53912.js:1:1892448
(anonymous function) — main.4e36a1fc2ea53912.js:1:1897113
(anonymous function) — main.4e36a1fc2ea53912.js:1:1903613
next — main.4e36a1fc2ea53912.js:1:1892448
(anonymous function) — main.4e36a1fc2ea53912.js:1:1905747
(anonymous function) — main.4e36a1fc2ea53912.js:1:1903613
next — main.4e36a1fc2ea53912.js:1:1892448
(anonymous function) — main.4e36a1fc2ea53912.js:1:1907215
(anonymous function) — main.4e36a1fc2ea53912.js:1:1903613
next — main.4e36a1fc2ea53912.js:1:1892448
_subscribe — main.4e36a1fc2ea53912.js:1:1887508
_trySubscribe — main.4e36a1fc2ea53912.js:1:1888252
(anonymous function) — main.4e36a1fc2ea53912.js:1:1888200
c — main.4e36a1fc2ea53912.js:1:1919888
subscribe — main.4e36a1fc2ea53912.js:1:1888103
(anonymous function) — main.4e36a1fc2ea53912.js:1:1888178
c — main.4e36a1fc2ea53912.js:1:1919888
subscribe — main.4e36a1fc2ea53912.js:1:1888103
(anonymous function) — main.4e36a1fc2ea53912.js:1:1907194
(anonymous function) — main.4e36a1fc2ea53912.js:1:1921682
(anonymous function) — main.4e36a1fc2ea53912.js:1:1888155
c — main.4e36a1fc2ea53912.js:1:1919888
subscribe — main.4e36a1fc2ea53912.js:1:1888103
(anonymous function) — main.4e36a1fc2ea53912.js:1:1905689
(anonymous function) — main.4e36a1fc2ea53912.js:1:1921682
(anonymous function) — main.4e36a1fc2ea53912.js:1:1888155
c — main.4e36a1fc2ea53912.js:1:1919888
subscribe — main.4e36a1fc2ea53912.js:1:1888103
(anonymous function) — main.4e36a1fc2ea53912.js:1:1897068
M — main.4e36a1fc2ea53912.js:1:1897200
(anonymous function) — main.4e36a1fc2ea53912.js:1:1897016
M — main.4e36a1fc2ea53912.js:1:1897200
(anonymous function) — main.4e36a1fc2ea53912.js:1:1896942
_trySubscribe — main.4e36a1fc2ea53912.js:1:1888252
(anonymous function) — main.4e36a1fc2ea53912.js:1:1888200
c — main.4e36a1fc2ea53912.js:1:1919888
subscribe — main.4e36a1fc2ea53912.js:1:1888103
(anonymous function) — main.4e36a1fc2ea53912.js:1:1907194
(anonymous function) — main.4e36a1fc2ea53912.js:1:1921682
(anonymous function) — main.4e36a1fc2ea53912.js:1:1888155
c — main.4e36a1fc2ea53912.js:1:1919888
subscribe — main.4e36a1fc2ea53912.js:1:1888103
(anonymous function) — main.4e36a1fc2ea53912.js:1:1905916
(anonymous function) — main.4e36a1fc2ea53912.js:1:1921682
(anonymous function) — main.4e36a1fc2ea53912.js:1:1888155
c — main.4e36a1fc2ea53912.js:1:1919888
subscribe — main.4e36a1fc2ea53912.js:1:1888103
(anonymous function) — main.4e36a1fc2ea53912.js:1:1907194
(anonymous function) — main.4e36a1fc2ea53912.js:1:1921682
(anonymous function) — main.4e36a1fc2ea53912.js:1:1888155
c — main.4e36a1fc2ea53912.js:1:1919888
subscribe — main.4e36a1fc2ea53912.js:1:1888103
(anonymous function) — main.4e36a1fc2ea53912.js:1:1907194
(anonymous function) — main.4e36a1fc2ea53912.js:1:1921682
(anonymous function) — main.4e36a1fc2ea53912.js:1:1888155
c — main.4e36a1fc2ea53912.js:1:1919888
subscribe — main.4e36a1fc2ea53912.js:1:1888103
(anonymous function) — main.4e36a1fc2ea53912.js:1:1897068
M — main.4e36a1fc2ea53912.js:1:1897200
(anonymous function) — main.4e36a1fc2ea53912.js:1:1897016
M — main.4e36a1fc2ea53912.js:1:1897200
(anonymous function) — main.4e36a1fc2ea53912.js:1:1896942
_trySubscribe — main.4e36a1fc2ea53912.js:1:1888252
(anonymous function) — main.4e36a1fc2ea53912.js:1:1888200
c — main.4e36a1fc2ea53912.js:1:1919888
subscribe — main.4e36a1fc2ea53912.js:1:1888103
(anonymous function) — main.4e36a1fc2ea53912.js:1:1907194
(anonymous function) — main.4e36a1fc2ea53912.js:1:1921682
(anonymous function) — main.4e36a1fc2ea53912.js:1:1888155
c — main.4e36a1fc2ea53912.js:1:1919888
subscribe — main.4e36a1fc2ea53912.js:1:1888103
(anonymous function) — main.4e36a1fc2ea53912.js:1:1911414
(anonymous function) — main.4e36a1fc2ea53912.js:1:1903613
next — main.4e36a1fc2ea53912.js:1:1892448
(anonymous function) — main.4e36a1fc2ea53912.js:1:1903613
next — main.4e36a1fc2ea53912.js:1:1892448
(anonymous function) — main.4e36a1fc2ea53912.js:1:1907215
(anonymous function) — main.4e36a1fc2ea53912.js:1:1903613
next — main.4e36a1fc2ea53912.js:1:1892448
(anonymous function) — main.4e36a1fc2ea53912.js:1:1907215
(anonymous function) — main.4e36a1fc2ea53912.js:1:1903613
next — main.4e36a1fc2ea53912.js:1:1892448
(anonymous function) — main.4e36a1fc2ea53912.js:1:1903613
next — main.4e36a1fc2ea53912.js:1:1892448
(anonymous function) — main.4e36a1fc2ea53912.js:1:1907215
(anonymous function) — main.4e36a1fc2ea53912.js:1:1903613
next — main.4e36a1fc2ea53912.js:1:1892448
(anonymous function) — main.4e36a1fc2ea53912.js:1:1897113
(anonymous function) — main.4e36a1fc2ea53912.js:1:1903613
	handleError (main.4e36a1fc2ea53912.js:1:2190754)
	next (main.4e36a1fc2ea53912.js:1:2301833)
	next (main.4e36a1fc2ea53912.js:1:1893068)
	_next (main.4e36a1fc2ea53912.js:1:1892752)
	next (main.4e36a1fc2ea53912.js:1:1892448)
	(anonymous function) (main.4e36a1fc2ea53912.js:1:1890410)
	c (main.4e36a1fc2ea53912.js:1:1919888)
	next (main.4e36a1fc2ea53912.js:1:1890245)
	emit (main.4e36a1fc2ea53912.js:1:2191786)
	run (polyfills.a1b1172df5a5f0a4.js:1:25438)
	onHandleError (main.4e36a1fc2ea53912.js:1:2229518)
	runTask (polyfills.a1b1172df5a5f0a4.js:1:26183)
	invokeTask (polyfills.a1b1172df5a5f0a4.js:1:32046)
```
## To Reproduce
You can reproduce on https://demo.dspace.org
1. Login as admin
2. Go to System Wide Alert, fill in a system wide alert, enable the alert, click save
3. Open the site in another browser, or logout and reload
4. There is no alert displayed
5. You might get an error message on the browser console

## Expected behavior
The system wide alert should display reliable when turned on.

**Comment by atarix83** [comment]
@tdonohue 
we can work on this one

**Comment by MMilosz** [comment]
I'm reopening this because there's still a related problem that affects System-Wide Alert visibility. To clarify, **the original issue** (alert suddenly disappearing or not showing at all) currently **appears to be resolved**. However, this part of the issue is still not fixed:
> but when you check in another browser or [...] it will not be visible

**Steps to reproduce** (works on demo.dspace.org, sandbox.dspace.org):
1. Open two different browsers (or one regular window and one incognito).
2. In Browser 1, open DSpace home page (or any page).
3. In Browser 2, log in as an admin, go to the System-Wide Alert settings, enable the alert, and save.
4. In Browser 1, without logging in, navigate through the site. Visit different pages, go back etc., but do not refresh the page.
5. Observe the **alert never appears in Browser 1**.

This is particularly a problem when the alert is needed for real-time updates, like urgent maintenance or ongoing issues

**Comment by tdonohue** [comment]
@MMilosz : Please create a new ticket for the related issue.  By reopening older tickets, it can get confusing to understand which versions of DSpace a specific issue was fixed in.  For instance, this ticket is marked as fixed in 8.1 (which is accurate).  If we change it to be fixed in a different version, we will no longer be able to easily track which parts were fixed in 8.1 and which were fixed later (as GitHub only allows for one "milestone" to be assigned to a ticket).

In other words, please refrain from reopening old tickets.  Instead, the best practice is to create a *new* ticket and link back to the older one.  That way the new ticket can have a different "milestone" and even be assigned to a different developer .

Closing this again as the original issue was fixed in 8.1.  A new ticket should be created for any related issues that still exist.
